### PR TITLE
feat: [TKC-1887]add on prem installer as the testkube command

### PR DESCRIPTION
**Pull request by nicufk**
## Checklist (choose whats happened)

- [ ] breaking change! (describe)
- [ ] tested locally
- [ ] tested on cluster
- [ ] added new dependencies
- [ ] updated the docs
- [ ] added a test

## Breaking changes

-

## Changes

-

## Fixes

-

### CODE REVIEW

File: pkg/k8sclient/k8sclient.go [hunk 1]
```diff
@@ -268,3 +276,123 @@ func GetPodLogs(ctx context.Context, k8sClient kubernetes.Interface, namespace s
...
+	}
+	svc, err := clientSet.CoreV1().Services(namespace).Get(ctx, serviceName, v1.GetOptions{})
+	if err != nil {
+		return err
+	}
+
+	var podPort intstr.IntOrString
+	for _, port := range svc.Spec.Ports {
+		if port.Port == int32(servicePort) {
+			podPort = port.TargetPort
```
Comment (vsukhin) [inline pkg/k8sclient/k8sclient.go:294]: break

File: pkg/k8sclient/k8sclient.go [hunk 2]
```diff
@@ -268,3 +276,123 @@ func GetPodLogs(ctx context.Context, k8sClient kubernetes.Interface, namespace s
...
+
+	if servicePod == nil {
+		return fmt.Errorf("no running pods found for service %s/%s", namespace, serviceName)
+	}
+
+	var podPortNumber int32
+	for _, c := range servicePod.Spec.Containers {
+		for _, p := range c.Ports {
+			if p.ContainerPort == podPort.IntVal || p.Name == podPort.StrVal {
+				podPortNumber = p.ContainerPort
```
Comment (vsukhin) [inline pkg/k8sclient/k8sclient.go:325]: break

File: pkg/k8sclient/k8sclient.go [hunk 3]
```diff
@@ -268,3 +276,123 @@ func GetPodLogs(ctx context.Context, k8sClient kubernetes.Interface, namespace s
...
+		CoreV1().
+		RESTClient().
+		Post().
+		Resource("pods").
+		Namespace(namespace).
+		Name(servicePod.Name).
+		SubResource("portforward").
+		URL()
+
+	dialer := spdy.NewDialer(upgrader, &http.Client{Transport: transport}, "POST", url)
```
Comment (vsukhin) [inline pkg/k8sclient/k8sclient.go:352]: http.POST

File: pkg/k8sclient/k8sclient.go [hunk 4]
```diff
@@ -268,3 +276,123 @@ func GetPodLogs(ctx context.Context, k8sClient kubernetes.Interface, namespace s
...
+	forwarder, err := portforward.New(dialer, []string{fmt.Sprintf("%d:%d", localhostPort, podPortNumber)}, ctx.Done(), readyChan, os.Stdout, os.Stderr)
```
Comment (vsukhin) [inline pkg/k8sclient/k8sclient.go:353]: why do we port forward for a particular pod, not a service?

Comment (vsukhin) [reply pkg/k8sclient/k8sclient.go:353]: should be localhostPort fixed, or we can dynamically check available one?

File: cmd/kubectl-testkube/commands/init.go [hunk 5]
```diff
@@ -59,3 +82,107 @@ func NewInitCmd() *cobra.Command {
...
+		Run: func(cmd *cobra.Command, args []string) {
+			ui.Info("WELCOME TO")
+			ui.Logo()
+
+			ui.NL()
+			if license == "" {
+				ui.Warn("License key is required to install " + demoInstallationName)
+				return
+			}
+			/*
```
Comment (vsukhin) [inline cmd/kubectl-testkube/commands/init.go:103]: why it's commented?

**Review by vsukhin — COMMENTED** [review]
a few questions

File: pkg/k8sclient/k8sclient.go (hunk 4 above)
Comment (nicufk) [reply pkg/k8sclient/k8sclient.go:355]: I was not able to make it work with the service

Comment (vsukhin) [reply pkg/k8sclient/k8sclient.go]: we can go with it, but pod is not something relaible, can be esily killed and restrarted. service is a logical unit

**Pull request merged**